witness/src/lib.rs

```diff
@@ -41,14 +47,20 @@ impl<T: Sized + Sync + Clone + Send + Copy + Default + PrimeCharacteristicRing>
     where
         Standard: Distribution<T>,
     {
+        debug_assert!(rows > 0);
+        let num_row_padded = next_pow2_instance_padding(rows);
         Self {
-            inner: p3_matrix::dense::RowMajorMatrix::rand(rng, rows, cols),
+            inner: p3_matrix::dense::RowMajorMatrix::rand(rng, num_row_padded, cols),
+            num_rows: rows,
+            is_padded: true,
             padding_strategy: InstancePaddingStrategy::Default,
```

Comment:
It seems that random matrices will not be padded with Default as suggested by the strategy field (but by random elements). I don't know if there are situations where this is a problem.

Reply:
Good catch! it will not be a problem for now, but to make api semantically correct, I modify the logic to make padding part fill with default